include/utility/dbus_utility.hpp

```diff
@@ -286,5 +286,34 @@ inline types::BiosAttributeCurrentValue
 
     return std::get<1>(l_attributeVal);
 }
+
+/**
+ * @brief API to check if Chassis is powered on.
+ *
+ * This API queries Phosphor Chassis State Manager to know whether
+ * Chassis is powered on.
+ *
+ * @throw std::runtime_error
+ * @return true if chassis is powered on, false otherwise
+ */
+inline bool isChassisPowerOn()
+{
+    auto powerState = dbusUtility::readDbusProperty(
+        "xyz.openbmc_project.State.Chassis",
+        "/xyz/openbmc_project/state/chassis0",
+        "xyz.openbmc_project.State.Chassis", "CurrentPowerState");
+
+    if (auto curPowerState = std::get_if<std::string>(&powerState))
+    {
+        if ("xyz.openbmc_project.State.Chassis.PowerState.On" == *curPowerState)
+        {
+            logging::logMessage("VPD cannot be read in power on state.");
```

Comment:
@SunnySrivastava1984 , should we remove this log message as it is not specific to the functionality of this API?

Reply:
Can be modified, "Chassis is in on state"?

Reply:
Addressed this.